### PR TITLE
folly: Add FOLLY_HAVE_INT128_T compilation option

### PR DESCRIPTION
**Pull request by viirya**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/folly.rb [hunk 1]
```diff
@@ -55,6 +55,8 @@ def install
         -DFOLLY_USE_JEMALLOC=OFF
       ]
 
+      args += %w[-DFOLLY_HAVE_INT128_T=ON] if OS.mac?
```
Comment (carlocab) [inline Formula/folly.rb:58]: Could you add a comment documenting what this is needed for?

Also, is this flag portable to ARM macOS?

**Review by carlocab — COMMENTED** [review]
Thanks, @viirya.

Please use the preferred commit-message style for homebrew/core. We put the name of the formula first in commit-message headings.

For new formulae:

> At Homebrew, we like to put the name of the formula up front like so: `foobar 7.3 (new formula)`.

For existing formulae:

> The preferred commit message format for simple version updates is `foobar 7.3` and for fixes is `foobar: fix flibble matrix.`.

Refer to the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit) for more details. Also, when making further changes to your pull request, adhere to the following guidelines to make sure that @BrewTestBot can merge your commits:

> - One formula per commit; one commit per formula.
> - Keep merge commits out of the pull request.

File: Formula/folly.rb (hunk 1 above)
Comment (viirya) [reply Formula/folly.rb:58]: As not all targets support it, this is not default. As I searched, not all compiler versions have the way to automatically detect it. I think it is why Folly doesn't do it.

**Comment by viirya** [comment]
> One formula per commit; one commit per formula.
> Keep merge commits out of the pull request.

Not sure if I understand it correctly. So my commit is only for Folly formula, it should satisfy first guideline?
Does second guideline mean not to do `git merge`?

File: Formula/folly.rb (hunk 1 above)
Comment (carlocab) [reply Formula/folly.rb:59]: Sorry, I should've been clearer: it is clear that this flag enables 128-bit integer features. What isn't clear at all is why we would want to do this, and that's what the comment needs to address.

Comment (viirya) [reply Formula/folly.rb:59]: On the targets where 128-bit integer are available, the caller of Folly library can call its methods with 128-bit integers. In the case, if we don't enable this option with Folly, there will be compilation error like "error: no matching function for call to ...". For example the [issue](https://github.com/facebookincubator/velox/issues/360) we encountered when building Velox.

Comment (viirya) [reply Formula/folly.rb:59]: Does it make sense for you?

Comment (SMillerDev) [reply Formula/folly.rb:59]: So why isn't this enabled by default? Why do we want to deviate from the settings the Folly devs chose here?

**Pull request closed without merging**